hypershift-operator/controllers/nodepool/kubevirt.go

```diff
@@ -1,15 +1,36 @@
 package nodepool
 
 import (
+	"strings"
+
 	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
 	capikubevirt "sigs.k8s.io/cluster-api-provider-kubevirt/api/v1alpha1"
 )
 
-func kubevirtMachineTemplateSpec(nodePool *hyperv1.NodePool) *capikubevirt.KubevirtMachineTemplateSpec {
+func kubevirtMachineTemplateSpec(nodePool *hyperv1.NodePool, containerDiskChecksum string) *capikubevirt.KubevirtMachineTemplateSpec {
+	template := nodePool.Spec.Platform.Kubevirt.NodeTemplate.DeepCopy()
+	dataVolumeTemplates := template.Spec.DataVolumeTemplates
+	for i, dv := range dataVolumeTemplates {
+		if dv.Name == "containervolume" {
+			if dv.Spec.Source != nil && dv.Spec.Source.Registry != nil &&
+				dv.Spec.Source.Registry.URL != nil && strings.Contains(*dv.Spec.Source.Registry.URL, "{rhcos:version}") {
+				image := strings.ReplaceAll(*dv.Spec.Source.Registry.URL, "{rhcos:version}", containerDiskChecksum)
+				dataVolumeTemplates[i].Spec.Source.Registry.URL = &image
+			}
+		}
+	}
+	volumes := template.Spec.Template.Spec.Volumes
+	for i, volume := range volumes {
+		if volume.Name == "containervolume" {
+			if volume.ContainerDisk != nil && strings.Contains(volume.ContainerDisk.Image, "{rhcos:version}") {
+				volumes[i].ContainerDisk.Image = strings.ReplaceAll(volume.ContainerDisk.Image, "{rhcos:version}", containerDiskChecksum)
+			}
+		}
+	}
```

Comment:
I am undecided. At least `{rhcos:openstack:url}` probably makes sense, but I am not really a fan of http based import. I would prefer to import also into datavolumes via a container disk. The involved raw-conversion and issing integrity checks is something I really don't like about datavolumes in general.

Regarding to 

```
--containerdisk=quay.io/containerdisks/rhcos-{rhcos:major}.{rhcos:minor}:{rhcos:openstack:checksum}
```

Could you explain what the purpose of the major and minor template parameter is? I can understand `checksum` because it gives the exactly matching disk. If I want a specific image I can specify it with a clear tag and this would for instance mean that I don't know the version I install, but I assume that specific version a dedicated repo exists.